### PR TITLE
upgrade tsp compiler to 0.61

### PR DESCRIPTION
**Pull request by pshao25**
1. Upgrade tsp compiler to 0.61
2. Fix paging response issues by:
2.1 if user customize the paging response then use the customized mode.
2.2 if user defined model is the same as ResourceListResult, then replace it by ResourceListResult
2.3 if interface name conflicts with model name, then rename interface name
3. Support customize LRO
4. Support ArmProviderActionAsync template.

### CODE REVIEW

File: packages/extensions/openapi-to-typespec/src/interfaces.ts [hunk 1]
```diff
@@ -222,6 +222,18 @@ export interface TspArmResourceOperationBase extends WithDoc, WithFixMe, WithSup
   customizations?: string[];
 }
 
+export interface TspArmProviderActionOperation extends WithDoc, WithSummary {
+  kind: "ArmProviderActionAsync";
+  name: string;
+  action: string | undefined;
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/src/interfaces.ts:228]: can it just be `action?: string`?

File: packages/extensions/openapi-to-typespec/src/utils/type-mapping.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { isObjectSchema, Response, SchemaResponse } from "@autorest/codemodel";
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/src/utils/type-mapping.ts:1]: this is not in use

File: packages/extensions/openapi-to-typespec/src/generate/generate-operations.ts [hunk 3]
```diff
@@ -59,6 +71,55 @@ export function generateOperation(operation: TypespecOperation, operationGroup?:
...
+  const doc = generateDocs(operation);
+  const summary = generateSummary(operation);
+  const statements: string[] = [];
+  summary && statements.push(summary);
+  statements.push(doc);
+  const templateParameters = [];
+  const responses = [...new Set(operation.responses)];
+  // Workaround for array response, refactor later.
+  const response =
+    responses.length === 1 && responses[0].endsWith("[]") ? `{@body _: ${responses[0]}}` : responses.join("|");
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/src/generate/generate-operations.ts:84]: what if it is not array?

File: packages/extensions/openapi-to-typespec/src/interfaces.ts [hunk 4]
```diff
@@ -222,6 +222,18 @@ export interface TspArmResourceOperationBase extends WithDoc, WithFixMe, WithSup
   customizations?: string[];
 }
 
+export interface TspArmProviderActionOperation extends WithDoc, WithSummary {
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/src/interfaces.ts:225]: add a TODO to consolidate this with other options?

**Review by live1206 — APPROVED** [review]

File: packages/extensions/openapi-to-typespec/src/generate/generate-operations.ts (hunk 3 above)
Comment (pshao25) [reply packages/extensions/openapi-to-typespec/src/generate/generate-operations.ts:84]: Then there will not have the `{@body}` wrapper.

File: packages/extensions/openapi-to-typespec/src/interfaces.ts (hunk 1 above)
Comment (pshao25) [reply packages/extensions/openapi-to-typespec/src/interfaces.ts]: We can.

File: packages/extensions/openapi-to-typespec/src/interfaces.ts [hunk 5]
```diff
@@ -222,6 +222,19 @@ export interface TspArmResourceOperationBase extends WithDoc, WithFixMe, WithSup
   customizations?: string[];
 }
 
+// TO-DO: consolidate with other templates
+export interface TspArmProviderActionOperation extends WithDoc, WithSummary {
+  kind: "ArmProviderActionAsync";
+  name: string;
+  action?: string;
+  responses: string[] | undefined;
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/src/interfaces.ts:230]: there are more cases in the same contract

**Pull request merged**